W-11417123 Amazon SQS Use Cases (#2021)
mulesoft/docs-connectors: amazon-sqs/5.11/modules/ROOT/pages/index.adoc

== Common Use Cases for the Connector

Common use cases for Amazon SQS Connector include:

> 

Reviewer: Include a use case header and description for sending messages to a queue and retrieving messages from a queue as the first use case in the beginning
ex; 
Queue Messaging
Send messages to a queue and retrieve messages from a queue

* *Asynchronous Processing Patterns* +
Improve reliability of loosely-coupled architecture by implementing asynchronous messaging that buffers incoming payloads for consumption at the pace of the consuming services. 

> * *Decouple Microservices* +

Reviewer: ```suggestion
* *Microservice Decoupling* +
```

> Use Amazon SQS to build a set of loosely coupled systems that can evolve independently and flexibly by implementing one-way messaging pattern or the request-response messaging.

Reviewer: ```suggestion
Build a set of loosely coupled systems that can evolve independently and flexibly by implementing a one-way messaging pattern or a request-response messaging.
```

> * *Distribute Workloads* +

Reviewer: ```suggestion
* *Workload Distribution* +
```

> For distributed large workloads, use Amazon SQS to maintain a queue of all tasks, efficiently distribute the tasks between the machines or functions doing the work, and guarantee that every part of the work is only done once.

Reviewer: ```suggestion
For large workload distribution, maintain a queue of all tasks, efficiently distribute the tasks between the machines or functions doing the work, and guarantee that every part of the work is only done once.
```

> * *Schedule Batch jobs* +

Reviewer: ```suggestion
* *Batch Job Scheduling* +
```

> Leverage Amazon SQS connector to maintain a queue of jobs, handle retries, track job status, and so on.

Reviewer: ```suggestion
Maintain a queue of jobs, handle retries, track job status, and so on.
```

== Common Use Cases for the Connector

Common use cases for Amazon SQS Connector include:

* *Queue Messaging* +
Send messages to a queue and retrieve messages from a queue. 

* *Asynchronous Processing Patterns* +
Improve reliability of loosely-coupled architecture by implementing asynchronous messaging that buffers incoming payloads for consumption at the pace of the consuming services. 

* *Microservice Decoupling* +
Build a set of loosely coupled systems that can evolve independently and flexibly by implementing a one-way messaging pattern or a request-response messaging.

* *Workload Distribution* +

> For large workload distribution, maintain a queue of all tasks, efficiently distribute the tasks between the machines or functions doing the work, and guarantee that every part of the work is only done once.

Reviewer: ```suggestion
For large workload distribution, maintain a queue of all tasks, efficiently distribute the tasks between the machines or the functions doing the work, and guarantee that every part of the work is only done once.
```
